pyccel/complexity/arithmetic.py

```diff
@@ -4,87 +4,270 @@
+        else:
+            return ops + Symbol('SIN')
+
+    def _cost_NumpyCos(self, expr, **settings):
+        ops = sum(self._cost(i, **settings) for i in expr.args)
+        if self.mode:
+            return ops + 1
+        else:
+            return ops + Symbol('COS')
+
+    def _cost_NumpyTan(self, expr, **settings):
+        ops = sum(self._cost(i, **settings) for i in expr.args)
+        if self.mode:
+            return ops + 1
+        else:
+            return ops + Symbol('TAN')
+
+    def _cost_NumpyArcsin(self, expr, **settings):
+        ops = sum(self._cost(i, **settings) for i in expr.args)
+        if self.mode:
+            return ops + 1
+        else:
+            return ops + Symbol('ARCSIN')
+
+    def _cost_NumpyArccos(self, expr, **settings):
+        ops = sum(self._cost(i, **settings) for i in expr.args)
+        if self.mode:
+            return ops + 1
+        else:
+            return ops + Symbol('ARCCOS')
+
+    def _cost_NumpyArctan(self, expr, **settings):
+        ops = sum(self._cost(i, **settings) for i in expr.args)
+        if self.mode:
+            return ops + 1
+        else:
+            return ops + Symbol('ARCTAN')
+
+    def _cost_NumpyArctan2(self, expr, **settings):
+        ops = sum(self._cost(i, **settings) for i in expr.args)
+        if self.mode:
+            return ops + 1
+        else:
+            return ops + Symbol('ARCTAN2')
+
+    def _cost_NumpySinh(self, expr, **settings):
+        ops = sum(self._cost(i, **settings) for i in expr.args)
+        if self.mode:
+            return ops + 1
+        else:
+            return ops + Symbol('SINH')
+
+    def _cost_NumpyCosh(self, expr, **settings):
+        ops = sum(self._cost(i, **settings) for i in expr.args)
+        if self.mode:
+            return ops + 1
+        else:
+            return ops + Symbol('COSH')
+
+    def _cost_NumpyTanh(self, expr, **settings):
+        ops = sum(self._cost(i, **settings) for i in expr.args)
+        if self.mode:
+            return ops + 1
+        else:
+            return ops + Symbol('TANH')
+
+    def _cost_NumpyArcsinh(self, expr, **settings):
+        ops = sum(self._cost(i, **settings) for i in expr.args)
+        if self.mode:
+            return ops + 1
+        else:
+            return ops + Symbol('ARCSINH')
+
+    def _cost_NumpyArccosh(self, expr, **settings):
+        ops = sum(self._cost(i, **settings) for i in expr.args)
+        if self.mode:
+            return ops + 1
+        else:
+            return ops + Symbol('ARCCOSH')
+
+    def _cost_NumpyArctanh(self, expr, **settings):
+        ops = sum(self._cost(i, **settings) for i in expr.args)
+        if self.mode:
+            return ops + 1
+        else:
+            return ops + Symbol('ARCTANH')
+
+    def _cost_PyccelAdd(self, expr, **settings):
+        ops = sum(self._cost(i, **settings) for i in expr.args)
+        if self.mode:
+            return ops + 1
+        else:
+            return ops + ADD
```

Comment:
I see that you are creating `Symbol` objects for the Numpy classes, but you are using the same `Symbol` object for basic operators. I assume this is done to reduce memory consumption for functions that you don't think will be used often and reduce calculation time for operators which are likely to be used a lot?

Reply:
Yes of course, I just wondered whether there was a reason to create a symbol as in line 192 for some cases but have one symbol for the whole file (e.g `ADD` used in line 199). Presumably this is a question that only Ahmed can answer. Normally I would prefer to do the same thing throughout the file rather than using different strategies in different places.

But if creating `Symbol`s is slow then I can see my explanation above being a possible excuse for having 2 strategies

Reply:
Oh right, I understand now